app/process/watcher/prometheus/watcher.go

```diff
@@ -0,0 +1,461 @@
+	"github.com/limechain/hedera-eth-bridge-validator/config"
+	"github.com/limechain/hedera-eth-bridge-validator/constants"
+	"github.com/prometheus/client_golang/prometheus"
+	log "github.com/sirupsen/logrus"
+	"math/big"
+	"strconv"
+	"time"
+)
+
+type Watcher struct {
+	dashboardPolling          time.Duration
+	mirrorNode                client.MirrorNode
+	EVMClients                map[uint64]client.EVM
+	configuration             config.Config
+	prometheusService         service.Prometheus
+	logger                    *log.Entry
+	payerAccountBalanceGauge  prometheus.Gauge
+	bridgeAccountBalanceGauge prometheus.Gauge
+	operatorBalanceGauge      prometheus.Gauge
+	// A mapping, storing all network ID - asset address - metric name
+	assetsMetrics map[uint64]map[string]string
+}
+
+func NewWatcher(
+	dashboardPolling time.Duration,
+	mirrorNode client.MirrorNode,
+	configuration config.Config,
+	prometheusService service.Prometheus,
+	EVMClients map[uint64]client.EVM,
+) *Watcher {
+
+	var (
+		payerAccountBalanceGauge  prometheus.Gauge
+		bridgeAccountBalanceGauge prometheus.Gauge
+		operatorBalanceGauge      prometheus.Gauge
+		// A mapping, storing all network ID - asset address - metric name
+		assetsMetrics = make(map[uint64]map[string]string)
+	)
+
+	if prometheusService.GetIsMonitoringEnabled() {
+		payerAccountBalanceGauge = prometheusService.CreateGaugeIfNotExists(prometheus.GaugeOpts{
+			Name: constants.FeeAccountAmountGaugeName,
+			Help: constants.FeeAccountAmountGaugeHelp,
+			ConstLabels: prometheus.Labels{
+				constants.AccountMetricLabelKey: configuration.Bridge.Hedera.PayerAccount,
+			},
+		})
+		bridgeAccountBalanceGauge = prometheusService.CreateGaugeIfNotExists(prometheus.GaugeOpts{
+			Name: constants.BridgeAccountAmountGaugeName,
+			Help: constants.BridgeAccountAmountGaugeHelp,
+			ConstLabels: prometheus.Labels{
+				constants.AccountMetricLabelKey: configuration.Bridge.Hedera.BridgeAccount,
+			},
+		})
+		operatorBalanceGauge = prometheusService.CreateGaugeIfNotExists(prometheus.GaugeOpts{
+			Name: constants.OperatorAccountAmountName,
+			Help: constants.OperatorAccountAmountHelp,
+			ConstLabels: prometheus.Labels{
+				constants.AccountMetricLabelKey: configuration.Node.Clients.Hedera.Operator.AccountId,
+			},
+		})
+	}
+
+	return &Watcher{
+		dashboardPolling:          dashboardPolling,
+		mirrorNode:                mirrorNode,
+		EVMClients:                EVMClients,
+		configuration:             configuration,
+		prometheusService:         prometheusService,
+		logger:                    config.GetLoggerFor(fmt.Sprintf("Prometheus Metrics Watcher on interval [%s]", dashboardPolling)),
+		payerAccountBalanceGauge:  payerAccountBalanceGauge,
+		bridgeAccountBalanceGauge: bridgeAccountBalanceGauge,
+		operatorBalanceGauge:      operatorBalanceGauge,
+		assetsMetrics:             assetsMetrics,
+	}
+}
+
+func (pw Watcher) Watch(q qi.Queue) {
+	// there will be no handler, so the q is to implement the interface
```

Comment:
if we add:
```go
if !pw.prometheusService.GetIsMonitoringEnabled() {
	return
}
```
it will not have to check later on whether monitoring is enabled. (we are stopping it before the watcher registration as well)

Reply:
the other checks in the watcher were removed